### PR TITLE
Render mobile receipt items as a bounded, expandable list

### PR DESCRIPTION
**Pull request by brookjordan**
## Summary by CodeRabbit

* **New Features**
  * Enhanced receipt item display with improved layout and visual metadata organization.
  * Added defensive error handling to gracefully manage missing user information.

* **Bug Fixes**
  * Improved item formatting and amount parsing to prevent crashes with invalid data.

* **Tests**
  * Added comprehensive test coverage for receipt item display and edge cases.

### CODE REVIEW

File: mobile/lib/receipts/widgets/receipt_item_items.dart [hunk 1]
```diff
@@ -33,177 +32,325 @@ class ReceiptItemItems extends StatefulWidget {
...
+          final List<FormItem> newItems = <FormItem>[...widget.items];
+          newItems.removeAt(index);
+
+          setState(() {
+            receiptModel.setItems(newItems);
           });
+        },
+      );
     }
     // TODO: need to fix new item data being wiped out when adding
```
[the remark this answers is not in the document]
Comment (brookjordan) [reply mobile/lib/receipts/widgets/receipt_item_items.dart:311]: The TODO at line 311 (and the one at line 1) both predate this PR — they came in with the initial mobile commit f2d6acc3. This PR is a layout fix, so leaving the form-state cleanup as separate scope.
